### PR TITLE
Migrate more icons to lucide-react

### PR DESCRIPTION
**Pull request by bduffany**
---

**Version bump**: Patch

### CODE REVIEW

File: app/root/root.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ export default class RootComponent extends React.Component {
       next: (user: User) => this.setState({ ...this.state, user }),
     });
     faviconService.setDefaultFavicon();
+    window._preferences = this.state.preferences;
```
Comment (siggisim) [inline app/root/root.tsx:44]: intentional?

File: enterprise/app/root/root.tsx [hunk 2]
```diff
@@ -70,6 +70,7 @@ export default class EnterpriseRootComponent extends React.Component {
     authService.register();
     router.register(this.handlePathChange.bind(this));
     faviconService.setDefaultFavicon();
+    (window as any)._preferences = this.state.preferences;
```
Comment (siggisim) [inline enterprise/app/root/root.tsx:73]: same as above

**Review by siggisim — APPROVED** [review]
Thanks for doing this!

After clicking around in dev, I found a couple of places that look a little funky - not sure if you want to handle that in this PR or a separate one (or maybe they're already taken care of).

Commit and branch icons on the history page are darker than others:
Screen Shot 2021-11-29 at 10 24 56 AM

The filter icon the raw tab is a little funky:
Screen Shot 2021-11-29 at 10 25 15 AM

File: app/root/root.tsx (hunk 1 above)
Comment (bduffany) [reply app/root/root.tsx:44]: Ah yes, I was using this to be able to quickly switch between dense and non-dense on a given page. Was calling `_preferences.toggleDenseMode()` in the console

**Comment by bduffany** [comment]
> Thanks for doing this!
> 
> After clicking around in dev, I found a couple of places that look a little funky - not sure if you want to handle that in this PR or a separate one (or maybe they're already taken care of).
> 
> Commit and branch icons on the history page are darker than others: Screen Shot 2021-11-29 at 10 24 56 AM
> 
> The filter icon the raw tab is a little funky: Screen Shot 2021-11-29 at 10 25 15 AM

The filter icons are fixed in this PR -- I'll take another look at the git icons in a separate PR

**Pull request merged**